tomviz/vtkChartTransfer2DEditor.cxx

```diff
@@ -48,14 +51,48 @@ vtkStandardNewMacro(vtkChartTransfer2DEditor)
 
 vtkChartTransfer2DEditor::~vtkChartTransfer2DEditor() = default;
 
-void vtkChartTransfer2DEditor::SetTransfer2D(vtkImageData* transfer2D)
+void vtkChartTransfer2DEditor::SetTransfer2D(vtkImageData* transfer2D,
+                                             vtkRectd* box)
 {
+  // Set the box first, GenerateTransfer2D writes to it
+  // so avoid writing to the old one.
+  if (box != nullptr) {
+    this->Transfer2DBox = box;
+  } else {
+    // Make sure to use internal dummy box.  This is set to
+    // nullptr when the active datasource/module is cleared
+    // which usually means it was deleted.  Avoid pointers to
+    // deleted data by setting this to a local dummy value.
+    this->Transfer2DBox = &this->DummyBox;
```

Comment:
The vtkRect classes are simple, and should generally be setting the values of the object, rather than passing around/setting the pointer to a single object. It is just 4 doubles, and the use of pointers looks overly complex when a simpler set/get paradigm could be used avoiding the headaches of memory management.

Reply:
The widget has to be able to edit the box in the DataSource/Module the same way it edits the transfer function.  Thus the pointers.  I think they should all be wrapped in a transfer function 2D object, but you didn't like that approach.

I could change it to emit some vtk event when the box changes and hook a Qt signal to that that updates the other box.  But managing what gets updated as the user changes the active objects will be just as complex as the pointers.